src/components/MaterialUI/DataDisplay/Table.js

```diff
@@ -1,64 +1,361 @@
+		padding: "17px 16px",
+		width: "50px",
+	},
+	headerCell: {
+		padding: "11px 16px",
+		textAlign: "left",
+		fontWeight: theme.typography.fontWeightBold,
+	},
+	headerCellSelect: {
+		padding: "8px 16px",
+		width: "50px",
+		textAlign: "left",
+		fontWeight: theme.typography.fontWeightBold,
+	},
+	rowSelectCheckbox: {
+		padding: 0,
 	},
 }));
 
-const Table = ({ headers, rows, checkbox, placeholder }) => {
-	const classes = useStyles();
+function propsAreEqualRow(prev, next) {
+	return prev.selected === next.selected;
+}
+
+export const MemoTableRow = React.memo(TableRow, propsAreEqualRow);
+
+function propsAreEqualBody(prev, next) {
+	return (
+		next.contentTimestamp !== undefined &&
+		prev.contentTimestamp === next.contentTimestamp &&
+		prev.selectedNumber === next.selectedNumber
+	);
+}
+
+const TableBody = props => {
+	return <tbody className={props.className}>{props.rows}</tbody>;
+};
+
+export const MemoTableBody = React.memo(TableBody, propsAreEqualBody);
+
+const TableCell = withDeferredTooltip(
+	React.forwardRef((props, ref) => {
+		return (
+			<td className={props.className} ref={ref} {...props}>
+				{props.value}
+			</td>
+		);
+	}),
+);
+
+// We will assume for now that cell will never change value so we want to avoid rerendering as much as possible
+export const MemoTableCell = React.memo(TableCell, () => true);
+
+const buildRowCheckbox = (classes, key, selectionHandlers) => {
+	const rowSelected = selectionHandlers.isSelected(key);
+
+	return (
+		<td className={classes.tableCellSelect} selected={rowSelected}>
+			<CheckboxMui
+				className={classes.rowSelectCheckbox}
+				checked={rowSelected}
+				onChange={event => selectionHandlers.selectionHandler(event, key)}
+			/>
+		</td>
+	);
+};
+
+const buildTableCheckbox = (classes, tableSelectionStatus, selectionMethods) => {
+	return (
+		<th key="tableSelector" className={classes.headerCellSelect}>
+			<CheckboxMui
+				className={classes.rowSelectCheckbox}
+				checked={tableSelectionStatus === tableSelectionMode.all}
+				indeterminate={tableSelectionStatus === tableSelectionMode.indeterminate}
+				onChange={event => selectionMethods.selectionHandler(event, null)}
+			/>
+		</th>
+	);
+};
+
+const StickerTableHeader = React.forwardRef((props, refHeader) => (
+	<div key="stickyHeader" className={props.classes.stickyHeaderHead}>
+		<TableMui className={props.classes.stickyHeaderTable}>
+			<TableHead ref={refHeader}>
+				<TableRow>{props.tableHeaders}</TableRow>
+			</TableHead>
+		</TableMui>
+		<div className={props.classes.stickyHeaderTableScroll} />
+	</div>
+));
+
+const buildTableHeaders = (
+	headers,
+	classes,
+	selectMode,
+	tableSelectionStatus,
+	selectionMethods,
+) => {
+	const tableCheckbox =
+		selectMode === true
+			? [buildTableCheckbox(classes, tableSelectionStatus, selectionMethods)]
+			: [];
+
+	return tableCheckbox.concat(
+		headers.map((header, index) => (
+			<th key={index} className={classes.headerCell} style={header.cellStyle}>
```

Comment:
What style property will do here? it's not react right?
You can use several classes inside className. To mix classes you can use classnames package (you can find sample inside ExpansionPanel.js)

Reply:
"style" is a react attribute yes, I wanted to include in column definitions (useConfig pattern) a way to provide custom style for a column. I will change my recipies for that.